imi/src/main/java/org/motechproject/nms/imi/service/impl/CdrFileServiceImpl.java

```diff
@@ -52,36 +72,93 @@
     private EventRelay eventRelay;
     private FileAuditRecordDataService fileAuditRecordDataService;
     private AlertService alertService;
-
+    private CsrValidatorService csrValidatorService;
 
 
     @Autowired
     public CdrFileServiceImpl(@Qualifier("imiSettings") SettingsFacade settingsFacade, EventRelay eventRelay,
-                              FileAuditRecordDataService fileAuditRecordDataService, AlertService alertService) {
+                              FileAuditRecordDataService fileAuditRecordDataService, AlertService alertService,
+                              CsrValidatorService csrValidatorService) {
         this.settingsFacade = settingsFacade;
         this.eventRelay = eventRelay;
         this.fileAuditRecordDataService = fileAuditRecordDataService;
         this.alertService = alertService;
+        this.csrValidatorService = csrValidatorService;
     }
 
 
-    private void reportErrorAndThrow(String file, String error) {
+    private void reportAuditAndThrow(String file, String error) {
         LOGGER.error(error);
         alertService.create(file, "CDR Detail File", error, AlertType.CRITICAL, AlertStatus.NEW, 0, null);
-        fileAuditRecordDataService.create(new FileAuditRecord(FileType.CDR_DETAIL_FILE, file, error, null, null));
+        fileAuditRecordDataService.create(new FileAuditRecord(FileType.CDR_DETAIL_FILE, file, false, error, null,
+                null));
         throw new IllegalStateException(error);
     }
 
 
-    // Aggregate all detail records for a requestId into a single summary record
-    private void parseDetailRecord(CallDetailRecordDto cdr, Map<String, CallSummaryRecordDto> records) {
+    private void reportAuditAndThrow(String file, List<String> errors) {
+        for (String error : errors) {
+            LOGGER.error(error);
+            alertService.create(file, "CDR Detail File", error, AlertType.CRITICAL, AlertStatus.NEW, 0, null);
```

Comment:
I setup a maximum number of errors after which the file will be rejected entirely and no more errors will be logged/audited/alerted